src/native/managed/cdacreader/src/Contracts/Thread.cs

```diff
@@ -0,0 +1,52 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics.CodeAnalysis;
+
+namespace Microsoft.Diagnostics.DataContractReader.Contracts;
+
+// TODO: [cdac] Add other counts / threads
+public record struct ThreadStoreData(
+    int ThreadCount,
+    TargetPointer FirstThread);
+
+internal sealed class Thread
```

Comment:
comparing to the sketch in the design docs, there's a layer of indirection missing.  The design was roughly:

* each contract has an interface and an implementation
* there's a contract registry that has a property for every contract interface that is lazily initialized
* when one contract needs to refer to another, it asks the contract registry